### PR TITLE
Standard product check

### PR DESCRIPTION
**Pull request by binni979**
This is the pull request to solve [this](https://github.com/NASA-IMPACT/pyQuARC/issues/261) issue.

Overview:
In the pyQuarc, while running the collection, it showed the warning message like this:
Screenshot 2024-04-29 at 9 57 31 AM

Changes made:
Added a standard_product_check function in the custom_validator file according to the rule mapping.json and check message.json. Both JSON uses the standard_product_check function but the function was not created. 
Now, no error is shown.
Tested on this ' C1577484501-LARC_ASDC' (umm-c) collection to make sure the if the warning shows up.

### CODE REVIEW

File: pyQuARC/code/custom_validator.py [hunk 1]
```diff
@@ -119,6 +119,26 @@ def dif_standard_product_check(*field_values):
                     break
         return {"valid": validity, "value": value}
 
+    @staticmethod
+    def standard_product_check(*field_values):
+        """
```
Comment (xhagrg) [inline pyQuARC/code/custom_validator.py:124]: Why rename the method? What's the benefit of using one or the other method name? Can the older model name not be used?

Comment (xhagrg) [reply pyQuARC/code/custom_validator.py:124]: we already have a `dif_standard_product_check` method, no?

Comment (binni979) [reply pyQuARC/code/custom_validator.py:124]: Yes, we have `dif_standard_product_check`  that is used to check the extended metadata for DIF schema, but the `standard_product_check`  is used to check the `StandardProduct` key inside the collection for schema echo-c and umm-c. 
I thought, creating a new function solved the problem, but it shows the error again and now I changed the code which is different from the previous function `dif_standard_product_check` .

Comment (xhagrg) [reply pyQuARC/code/custom_validator.py:124]: If it's just checking values in a constrained list, don't we already have a check that does this? why start a new one?

Comment (slesaad) [reply pyQuARC/code/custom_validator.py:124]: it's not that simple, some of these "fields" are not regular string fields, they are a list and you gotta check each of the item in the list to see if it exists. i believe this is one such case. similarly, and formatting is different for dif and other types, hence I think a new check is needed.

Comment (binni979) [reply pyQuARC/code/custom_validator.py]: Thank you @slesaad didi. Worked on that.

Comment (slesaad) [reply pyQuARC/code/custom_validator.py]: Did you test it? Everything working well?

Comment (binni979) [reply pyQuARC/code/custom_validator.py]: I tested it on the "UMM-C record (C1577484501-LARC_ASDC)".  Yes, everything working well.

**Review by slesaad — APPROVED** [review]

**Pull request merged**